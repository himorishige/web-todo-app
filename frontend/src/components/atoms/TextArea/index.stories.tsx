/** @jsxImportSource @emotion/react */
import { ComponentStory, ComponentMeta } from '@storybook/react';
import TextArea from '.';

export default {
  title: 'WebToDoApp/Atoms/TextArea',
  component: TextArea,
} as ComponentMeta<typeof TextArea>;

const Template: ComponentStory<typeof TextArea> = (args) => <TextArea {...args} />;

export const Default = Template.bind({});
Default.storyName = 'Default';
Default.args = {
  placeholder: 'メモ',
  disabled: false,
};
