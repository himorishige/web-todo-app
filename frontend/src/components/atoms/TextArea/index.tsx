/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { memo } from 'react';
import { Path, UseFormRegister } from 'react-hook-form';

export type Props = {
  placeholder: string;
  defaultValue?: string;
  disabled?: boolean;
  label: Path<Inputs>;
  register: UseFormRegister<Inputs>;
  maxLength?: number;
  onChange?: () => void;
};

type Inputs = {
  taskName: string;
  taskMemo: string;
};

const TextArea: React.VFC<Props> = memo(
  ({
    placeholder,
    defaultValue,
    disabled = false,
    maxLength = 140,
    label,
    register,
    ...props
  }: Props) => {
    return (
      <textarea
        css={textAreaStyle}
        placeholder={placeholder}
        {...register(label)}
        disabled={disabled}
        maxLength={maxLength}
        defaultValue={defaultValue}
        data-testid="tasks-textarea"
      />
    );
  },
);

export default TextArea;

const textAreaStyle = css`
  width: 100%;
  height: 100%;
  min-height: 30vh;
  padding: 1rem;
  border: 2px solid var(--primary-color);
  border-radius: 4px;
  color: var(--primary-color);
  &:disabled {
    background: #f8f8f8;
  }
`;
