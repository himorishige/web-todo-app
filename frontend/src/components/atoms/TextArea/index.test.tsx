import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TextArea from '.';
import { matchers } from '@emotion/jest';

// Add the custom matchers provided by '@emotion/jest'
expect.extend(matchers);

const placeholder = '入力';
const label = 'taskMemo';
const register = jest.fn();

describe('Atoms/TextArea', () => {
  test('初期状態はPropsが正しく反映されている', () => {
    render(
      <TextArea placeholder={placeholder} defaultValue="メモ" label={label} register={register} />,
    );
    const element = screen.getByTestId('tasks-textarea') as HTMLTextAreaElement;
    expect(element.placeholder).toBe(placeholder);
    expect(element.value).toBe('メモ');
  });

  test('フィールドに1を入力したら1が表示される', () => {
    render(
      <TextArea placeholder={placeholder} defaultValue="メモ" label={label} register={register} />,
    );
    const element = screen.getByTestId('tasks-textarea') as HTMLTextAreaElement;
    fireEvent.change(element, { target: { value: '1' } });
    expect(element.value).toBe('1');
  });

  test('正しくレンダリングされている', () => {
    const target = renderer
      .create(
        <TextArea placeholder={placeholder} defaultValue="" label={label} register={register} />,
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });
});
