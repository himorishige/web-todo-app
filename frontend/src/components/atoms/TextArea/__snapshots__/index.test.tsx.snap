// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/TextArea 正しくレンダリングされている 1`] = `
<textarea
  className="css-1w1700i"
  data-testid="tasks-textarea"
  defaultValue=""
  disabled={false}
  maxLength={140}
  placeholder="入力"
/>
`;
